openmicroscopy/openmicroscopy
# Tag tree actions

This PR updates the web with the same Tag & Tag set functionality as Insight.

![screen shot 2016-03-08 at 16 33 26](https://cloud.githubusercontent.com/assets/900055/13608188/939b00fe-e54b-11e5-9fd8-062062a1ac67.png)

To test:
 - Browse tags tree
 - Creation of tags / tag sets should appear as orphans in tree, unless tag is created with tag set selected (should be created as child of tag set).
     - Create Tag and Tag set from toolbar buttons
     - Create Tag and Tag set from right-click menu
 - Cut, Copy and Paste all data types, from toolbar and Right-click menu:
    - Tag / Tagset
    - Dataset / Project
    - Image / Dataset
    - Plate / Screen
 - Drag and Drop all data types that can be Cut as above
 - Tagged objects can also be Copied from below Tag (NOT Cut, as in Insight)
 - Delete Tag & Tag set

NB: @pwalczysko, @bramalingam and I had a discussion about whether you should be able to Cut OR Copy orphaned objects E.g.:
 - Dataset in P/D/I tree
 - Orphaned Image in P/D/I tree
 - Tag in Tag tree

Generally agreed that you should be able to Copy these (not current policy in Insight or Web & not changed in this PR).
Re: this PR, we allow to Copy:
 - Tagged Project/Dataset/Image etc in Tag tree (where you are not Cutting the parent link)
This is conceptually similar to Copy of an Orphaned object (also not Cutting any parent link).
